client/src/app/users/user-list.component.spec.ts

```diff
@@ -146,10 +142,26 @@ describe('Misbehaving User List', () => {
   }));
 
   it('generates an error if we don\'t set up a UserListService', () => {
+    const mockedMethod = spyOn(userList, 'getUsersFromServer').and.callThrough();
     // Since calling either getUsers() or getUsersFiltered() return
     // Observables that then throw exceptions, we don't expect the component
     // to be able to get a list of users, and serverFilteredUsers should
     // be undefined.
-    expect(userList.serverFilteredUsers).toBeUndefined();
+    expect(userList.serverFilteredUsers)
+      .withContext('service can\'t give values to the list if it\'s not there')
+      .toBeUndefined();
+    //expect(fixture.detectChanges)
```

Comment:
Why is this code commented out. Should we remove it?

Reply:
Is this a change request? The review says you requested changes, but I'm not seeing where those are. I'm happy to have you delete (or to delete myself) this unused code if that would do it.

Reply:
Yeah, I was proposing that we remove that commented out code. I think that having commented out code (especially without some note explaining why it's there and commented out) is just confusing for the students. And it then gives the impression that it's OK for them to have loads of commented out code in their projects, which we don't want.

So if we remove this then I think this is good to go!